### ISSUE TITLE
laymansync module's module_spec is old

### ISSUE THREAD

**Issue by opncow** [leycec/raiagent#53]
I get the following when running eix-update:

```
~ # eix-update 
Reading Portage settings...
Building database (/var/cache/eix/portage.eix)...
[...]
[2] "raiagent" /usr/portage/local/layman/raiagent (cache: parse|ebuild*#metadata-md5#metadata-flat#assign)
     Reading category   2|167 (  1): app-admin...laymansync module's module_spec is old, missing attribute: 'sourcefile'.  Backward compatibility may be removed in the future.
File: /usr/lib64/python3.4/site-packages/portage/sync/modules/laymansync/__init__.py
laymansync module's module_spec is old, missing attribute: 'sourcefile'.  Backward compatibility may be removed in the future.
File: /usr/lib64/python3.4/site-packages/portage/sync/modules/laymansync/__init__.py
     Reading category 167|167 (100) Finished           
[...]     
Applying masks...
Calculating hash tables...
Writing database file /var/cache/eix/portage.eix...
Database contains 19485 packages in 167 categories.
```

I do not know what or if something is wrong, but still wanted to report it.

**Comment by leycec** [comment]
Thumbs up for this preemptive strike on a possible issue. :rocket: 

The following lines suggest a `laymansync` issue unrelated to this repository... but, of course, I _would_ say that. I don't particularly want to do any work, so I'm happy to blame Layman for this:

```
File: /usr/lib64/python3.4/site-packages/portage/sync/modules/laymansync/__init__.py
laymansync module's module_spec is old, missing attribute: 'sourcefile'.  Backward compatibility may be removed in the future.
```

If you haven't already, consider [migrating](https://wiki.gentoo.org/wiki/Project:Portage/Sync) from the old `/var/lib/layman/make.conf` file to the new `/etc/portage/repos.conf` directory. Specifically, ensure that you have `layman >= 2.3.0` with the optional `sync-plugin-portage` USE flag enabled.

If after doing so you're still stumbling into the same brickwall, consider submitting a bug at the [official Gentoo bug tracker](http://bugs.gentoo.org).

As this _could_ genuinely be a `raiagent` issue, <sup>_Stranger things have happened. Like my toupée._</sup> I'll leave this open for the moment.

**Comment by opncow** [comment]
Thanks for the suggestion! I already migrated to the new layout some time ago.

But you seem to be right. That day the message was _constantly_ shown below raiagent on eix-sync. That made me think that it is a problem with the raiagent overlay. Now on sync it appears below another one, so it really seems to be not related to raiagent. 

Sorry for opening the issue so fast. I usually do not do that. For the first time, however, I could not find anything on google about a (this) message besides the python code that creates it. And that unfortunately does not give me a clue what is going wrong. I just though there were some changes in layman that your overlay does not respect, yet.

**Comment by opncow** [comment]
Updated to latest layman (2.4.1-r1) and the warning is gone btw...

**Comment by leycec** [comment]
**Fist-pump!** Thanks for tackling this head-on. Isn't nice when `emerge @world` actually fixes all of your problems for once? :sun_with_face:
